### PR TITLE
Lin contributions

### PR DESCRIPTION
**Pull request by petroskayange**

### CODE REVIEW

File: app/controllers/api/v1/reports_controller.rb [hunk 1]
```diff
@@ -33,9 +33,16 @@ def malaria_report
     render json: stats
   end
 
-  def diagnosis
+  def diagnosis_ls
```
Comment (ntumbuka) [inline app/controllers/api/v1/reports_controller.rb]: You don't need to do this... There is a generic endpoint for reports:

```HTTP
GET /api/v1/programs/:program_id/reports/:report_name?start_date=YYYY-MM-DD...
```
The above triggers the method `find_report` in the right report_engine (in your case `opd_service/report_engine#find_report`). The `find_report` method will be passed the report's name and other parameters specified in the URL.

Example:

```ruby
# app/services/opd_service/report_engine.rb

class OPDService::ReportEngine
  def find_report(name:, start_date:, **kwargs)
    case name
    when 'diagnosis_ls' then run_diagnosis_report(start_date, kwargs[:end_date], kwargs[:other_param])
    else raise NotFoundError, "Unknown OPD report: #{name}"
  end
end
```

File: .idea/BHT-EMR-API.iml [hunk 2]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ntumbuka) [inline .idea/BHT-EMR-API.iml]: Remove .idea folder

File: app/controllers/api/v1/reports_controller.rb (hunk 1 above)
Comment (petroskayange) [reply app/controllers/api/v1/reports_controller.rb]: We have done that now

File: .idea/BHT-EMR-API.iml (hunk 2 above)
Comment (petroskayange) [reply .idea/BHT-EMR-API.iml]: we have deleted it

**Review by mwatha — COMMENTED** [review]
we already have an endpoint doing this: 

def patient_details_by_id
 ......
end 

Can you use  "api/v1/patient/<patient_id>"???

File: app/controllers/api/v1/patients_controller.rb [hunk 3]
```diff
@@ -225,6 +225,12 @@ def visit
     render json: service.fetch_full_visit(patient, program, params[:date])
   end
 
+  def patient_details_by_id
+    patient_id = params[:patient_id]
```
Comment (mwatha) [inline app/controllers/api/v1/patients_controller.rb]: I meant here: can you use "api/v1/patient/<patient_id>"

**Review by mwatha — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/patients_controller.rb (hunk 3 above)
Comment (dominickasanga) [reply app/controllers/api/v1/patients_controller.rb]: Am working on it

**Pull request closed without merging**